libgraph/src/PropertyGraph.cpp

```diff
@@ -964,6 +968,29 @@ katana::PropertyGraph::DeleteEdgeIndex(const std::string& column_name) {
   return KATANA_ERROR(katana::ErrorCode::NotFound, "edge index not found");
 }
 
+// todo(Chak-Pong) expose it to python side
+katana::Result<bool>
+katana::PropertyGraph::IsEdgeIndexPresent(const std::string& column_name) {
+  for (const auto& existing_index : edge_indexes_) {
+    if (existing_index->column_name() == column_name) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// todo(Chak-Pong) expose it to python side
+katana::Result<bool>
+katana::PropertyGraph::IsNodeIndexPresent(const std::string& column_name) {
+  for (auto it = node_indexes_.begin(); it != node_indexes_.end(); it++) {
+    if ((*it)->column_name() == column_name) {
+      node_indexes_.erase(it);
```

Comment:
This code will be removed so this is just a comment for future coding you do. This will delete indexes in a method that by it's name should be read-only. This would have been caught if you had made these methods `const` (as they should be). Please be more careful.